agent/dns_test.go

```diff
@@ -5829,16 +5829,12 @@ func TestDNS_NonExistentDC_RPC(t *testing.T) {
 		server = false
 	`)
 	defer c.Shutdown()
-	testrpc.WaitForLeader(t, s.RPC, "dc1")
 
 	// Join LAN cluster
 	addr := fmt.Sprintf("127.0.0.1:%d", s.Config.SerfPortLAN)
 	_, err := c.JoinLAN([]string{addr})
 	require.NoError(t, err)
-	retry.Run(t, func(r *retry.R) {
-		require.Len(r, s.LANMembers(), 2)
-		require.Len(r, c.LANMembers(), 2)
-	})
+	testrpc.WaitForTestAgent(t, c.RPC, "dc1")
```

Comment:
This seems to work, but I'm not confident it is the correct function to use. Is there a better option? 

The flake was `TestAgent.dns: rpc error: error=No known Consul servers`, so I guess we have to wait for the client to fully join the cluster and receive the RPC address of the servers (or something like that) ?

Reply:
`WaitForTestAgent` queries the catalog, which will only return once there is a leader so I think it does exactly what you wanted.